# Проект "Исследование надежности заемщиков"


- - - - - - - 
# Описание проекта

**Заказчик**— кредитный отдел банка. 

**Входные данные от банка** — статистика о платёжеспособности клиентов.

**Цель исследования** — разобраться влияет ли семейное положение, уровень дохода и количество детей клиента на факт погашения кредита в срок.

Данные по статистикe о платёжеспособности клиентов получены из файла Яндекс.Практикум. О качестве данных ничего не известно. Поэтому перед анализом понадобится обзор данных.

Необходимо проверить данные на ошибки, оценить их влияние на исследование, а так же найти возможность исправить найденные ошибки, заполнив пропуски медианными значениями, обработать данные с неявными дубликатами и артефактами (аномалиями), при необходимости, удалив некоторые из них.

Таким образом, **исследование пройдёт в четыре этапа**:

1. Обзор данных.

2. Предобработка данных.

2. Исследование данных, ответы на вопросы.

2. Формирование общих выводов.

***Важно: результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку***.


<h1>Содержание:<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Откроем-таблицу-и-изучим-общую-информацию-о-данных" data-toc-modified-id="Откроем-таблицу-и-изучим-общую-информацию-о-данных-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Откроем таблицу и изучим общую информацию о данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Удаление-пропусков" data-toc-modified-id="Удаление-пропусков-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Удаление пропусков</a></span></li><li><span><a href="#Обработка-аномальных-значений" data-toc-modified-id="Обработка-аномальных-значений-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Обработка аномальных значений</a></span></li><li><span><a href="#Удаление-пропусков-(продолжение)" data-toc-modified-id="Удаление-пропусков-(продолжение)-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Удаление пропусков (продолжение)</a></span></li><li><span><a href="#Изменение-типов-данных" data-toc-modified-id="Изменение-типов-данных-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Изменение типов данных</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>Обработка дубликатов</a></span></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-2.6"><span class="toc-item-num">2.6&nbsp;&nbsp;</span>Категоризация данных</a></span></li><li><span><a href="#Шаг-3.-Исследуйте-данные-и-ответьте-на-вопросы" data-toc-modified-id="Шаг-3.-Исследуйте-данные-и-ответьте-на-вопросы-2.7"><span class="toc-item-num">2.7&nbsp;&nbsp;</span>Шаг 3. Исследуйте данные и ответьте на вопросы</a></span><ul class="toc-item"><li><span><a href="#3.1-Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?" data-toc-modified-id="3.1-Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?-2.7.1"><span class="toc-item-num">2.7.1&nbsp;&nbsp;</span>3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?</a></span></li><li><span><a href="#3.2-Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?" data-toc-modified-id="3.2-Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?-2.7.2"><span class="toc-item-num">2.7.2&nbsp;&nbsp;</span>3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?</a></span></li><li><span><a href="#3.3-Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?" data-toc-modified-id="3.3-Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?-2.7.3"><span class="toc-item-num">2.7.3&nbsp;&nbsp;</span>3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?</a></span></li><li><span><a href="#3.4-Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?" data-toc-modified-id="3.4-Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?-2.7.4"><span class="toc-item-num">2.7.4&nbsp;&nbsp;</span>3.4 Как разные цели кредита влияют на его возврат в срок?</a></span></li><li><span><a href="#3.5-Приведите-возможные-причины-появления-пропусков-в-исходных-данных." data-toc-modified-id="3.5-Приведите-возможные-причины-появления-пропусков-в-исходных-данных.-2.7.5"><span class="toc-item-num">2.7.5&nbsp;&nbsp;</span>3.5 Приведите возможные причины появления пропусков в исходных данных.</a></span></li><li><span><a href="#3.6-Объясните,-почему-заполнить-пропуски-медианным-значением-—-лучшее-решение-для-количественных-переменных." data-toc-modified-id="3.6-Объясните,-почему-заполнить-пропуски-медианным-значением-—-лучшее-решение-для-количественных-переменных.-2.7.6"><span class="toc-item-num">2.7.6&nbsp;&nbsp;</span>3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.</a></span></li></ul></li><li><span><a href="#Шаг-4:-общий-вывод." data-toc-modified-id="Шаг-4:-общий-вывод.-2.8"><span class="toc-item-num">2.8&nbsp;&nbsp;</span>Шаг 4: общий вывод.</a></span></li></ul></li></ul></div>

## Откроем таблицу и изучим общую информацию о данных

**Задание 1. Импортируем библиотеку pandas. Считаем данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('C:/Users/Мария/Desktop/Практикум/Датасеты/data.csv')

**Задание 2. Выведем первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

,children,days_employed,dob_years,education,education_id,family_status,family_status_id,gender,income_type,debt,total_income,purpose
0,1,-8437.673028,42,высшее,0,женат / замужем,0,F,сотрудник,0,253875.639453,покупка жилья
1,1,-4024.803754,36,среднее,1,женат / замужем,0,F,сотрудник,0,112080.014102,приобретение автомобиля
2,0,-5623.422610,33,Среднее,1,женат / замужем,0,M,сотрудник,0,145885.952297,покупка жилья
3,3,-4124.747207,32,среднее,1,женат / замужем,0,M,сотрудник,0,267628.550329,дополнительное образование
4,0,340266.072047,53,среднее,1,гражданский брак,1,F,пенсионер,0,158616.077870,сыграть свадьбу
5,0,-926.185831,27,высшее,0,гражданский брак,1,M,компаньон,0,255763.565419,покупка жилья
6,0,-2879.202052,43,высшее,0,женат / замужем,0,F,компаньон,0,240525.971920,операции с жильем
7,0,-152.779569,50,СРЕДНЕЕ,1,женат / замужем,0,M,сотрудник,0,135823.934197,образование
8,2,-6929.865299,35,ВЫСШЕЕ,0,гражданский брак,1,F,сотрудник,0,95856.832424,на проведение свадьбы
9,0,-2188.756445,41,среднее,1,женат / замужем,0,M,сотрудник,0,144425.938277,покупка жилья для семьи


**Описание данных**

- `children` — количество детей в семье  
- `days_employed` — общий трудовой стаж в днях
- `dob_years` — возраст клиента в годах
- `education` — уровень образования клиента
- `education_id` — идентификатор уровня образования
- `family_status` — семейное положение
- `family_status_id` — идентификатор семейного положения
- `gender` — пол клиента
- `income_type` — тип занятости
- `debt` — имел ли задолженность по возврату кредитов
- `total_income` — ежемесячный доход
- `purpose` — цель получения кредита

**Задание 3. Выведем основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


**Вывод**

В каждой строке таблицы `data` — все необходимые банку сведения о его клиентах, собранные в процессе кредитования (доход, дети, цель кредита, трудовой стаж, семейное положение и прочее).

Предварительно можно утверждать, что данных достаточно для проведения анализа и ответов на заданные вопросы. Но в исходных данных (столбцы `days_employed` и `total_income`) встречаются пропуски.

Чтобы двигаться дальше, нужно устранить проблемы в данных.

## Предобработка данных

### Удаление пропусков

**Задание 4. Выведем количество пропущенных значений для каждого столбца. Используем комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце обработаем на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработаем значения в этом столбце: заменим все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведем медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставим их как есть.

**Задание 8. Выведем перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

**Задание 9. В столбце `children` есть два аномальных значения. Удалим строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведем перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков (продолжение)

**Задание 11. Заполним пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедимся, что все пропуски заполнены. Проверим себя и ещё раз выведем количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Заменим вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработаем неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведем их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведем на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалим их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создадим в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведем на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создадим функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используем собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучим данные в столбце `purpose` и определим, какие подстроки помогут правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

**Вывод**

**Предобработка обнаружила следующие проблемы в данных**:

- пропущенные значения в столбцах `days_employed` и `total_income`;
- дубликаты — явные и неявные;
- артефакты (аномалии) в данных.

**Вышеуказанные проблемы были решены следующим образом**:

1. Пропуски в столбце `total_income` были заполнены медианным доходом для каждой категории заемщиков. В столбце `days_employed` пропуски заполнили медианными значениями по каждому типу занятости `income_type`.

2. Обработали неявные дубликаты в столбце `education`, удалили явные. 

3. Заменили вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.

4. Обработали все артефакты (аномалии) данных в столбце `days_employed`: заменили все отрицательные значения положительными с помощью метода `abs()`.

5. В столбце `children` нашли два аномальных значения, удалили из датафрейма `data` строки, в которых встречаются такие аномальные значения.

**Далее всех клиентов поделили на категории в зависимости от:**

- уровня дохода, создав и добавив в датафрейм новый столбец `total_income_category`;
- от целей кредита, создав и добавив в датафрейм новый столбец `purpose_category`.
 
**Теперь можно перейти к исследованию данных и ответам на вопросы.**

### Шаг 3. Исследуйте данные и ответьте на вопросы

<a id='step_3'></a>

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Для определения *зависимости между количеством детей и возвратом кредита в срок*, нам понадобятся два показателя:

- `children` — количество детей в семье заемщика;
- `debt` — имел ли заемщик задолженность по возврату кредитов (`0` - нет, `1` - да).

Для повышения достоверности выводов, полученных из настоящего анализа, предлагается:
- добавить еще один показатель `overdue_loan_percentage` - *процент просроченной задолженности по кредиту по всем заемщикам в %*, который будет использоваться во всех заданиях, где нужно проследить зависимость между заданной группой заемщиков и уплатой кредитов в срок. При дальнейшей группировке заемщиков на требуемые категории, станет хорошо видно, какой процент просрочки среди всех платежей имеет та или иная анализируемая группа заемщиков, следовательно, данный показатель даст более четкую картину наличия или отсутсвия зависимости между заданной категорией заемщиков и возвратом кредита в срок;
- разбить всех заемщиков на группы, в зависимости от количества детей в семье и результаты добавить в новый столбец `children_group` - *статус семьи заемщика, в зависимости от количества детей*.

In [22]:
#добавляем к исходной таблице data столбец overdue_loan_percentage 
data['overdue_loan_percentage'] = (data['debt'] / data['debt'].count()) * 100

Проверяем, добавился ли новый столбец `overdue_loan_percentage`в общий датафрейм:

In [23]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21331 entries, 0 to 21524
Data columns (total 15 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   children                 21331 non-null  int64  
 1   days_employed            21331 non-null  float64
 2   dob_years                21331 non-null  int64  
 3   education                21331 non-null  object 
 4   education_id             21331 non-null  int64  
 5   family_status            21331 non-null  object 
 6   family_status_id         21331 non-null  int64  
 7   gender                   21331 non-null  object 
 8   income_type              21331 non-null  object 
 9   debt                     21331 non-null  int64  
 10  total_income             21331 non-null  int32  
 11  purpose                  21331 non-null  object 
 12  total_income_category    21331 non-null  object 
 13  purpose_category         21331 non-null  object 
 14  overdue_loan_percentag

Видим, что в строке с индексом 14 появился новый столбец `overdue_loan_percentage`.  Переходим к следующему этапу.               

С помощью функции `children_group()` разобьем всех заемщиков на группы, в зависимости от количества детей  в семье:

In [24]:
def children_group(children):
    """
    Возвращает статус семьи по значению количества у заемщика детей, используя правила:
    - 'бездетная семья' при значении children == 0;
    - 'малодетная семья' при значении children до двух человек;
    - 'многодетная семья' - во всех остальнных случаях (при значении children от 3 и выше)
    """
    if children == 0:
        return 'бездетная семья'
    
    if children <= 2:
        return 'малодетная семья'
    
    return 'многодетная семья'

# проверяем, правильно ли отвечает функция на переданные ей числа
children = 0
print(children_group(children))
children = 1
print(children_group(children))
children = 3
print(children_group(children))

# доавляем новый столбец children_group в data
data['children_group'] = data['children'].apply(children_group)

# выводим для проверки первые 10 строк столбца data['children_group']
print(data['children_group'].head(10))

бездетная семья
малодетная семья
многодетная семья
0     малодетная семья
1     малодетная семья
2      бездетная семья
3    многодетная семья
4      бездетная семья
5      бездетная семья
6      бездетная семья
7      бездетная семья
8     малодетная семья
9      бездетная семья
Name: children_group, dtype: object


Теперь сформируем по столбцам, необходимым нам для анализа ( `debt`, `overdue_loan_percentage`, `children_group`) сводную таблицу **data_pivot_children_debt** методом *pivot_table()*:

In [25]:
data_pivot_children_debt = data.pivot_table(index=['children_group'],\
                                            values = ['debt', 'overdue_loan_percentage'] , aggfunc='sum')

print() # пробел для лучшей визуалиции данных при выводе на экран
print('Анализ наличия зависимости между количеством детей и возвратом кредита в срок:')
display(data_pivot_children_debt)

#выведем на экран еще общее число заемщиков
print('Итого заемщиков:', data['children'].count())


Анализ наличия зависимости между количеством детей и возвратом кредита в срок:


,debt,overdue_loan_percentage
children_group,,
бездетная семья,1063,4.983358
малодетная семья,638,2.990952
многодетная семья,31,0.145328


Итого заемщиков: 21331


Согласно данным сводной таблицы `data_pivot_children_debt` прослеживается **обратная пропорциональная зависимость между количеством детей и возвратом кредита в срок**. 

Заемщики, которые в анкете указали на отсутвие детей, намного чаще имеют просроченную задолженность по кредиту, чем заемщики с детьми. Причем, процент невозврата кредита в срок напрямую зависит от количества детей должника: 

- у заемщиков **без детей** процент просрочки равен **4.98%** (или 1 063 случаев из всех анализируемых 21 331),
- у **малодетных заемщиков** с *одним или двумя детьми* - **2.99%** (638 случаев), 
- у **многодетных заемщиков** с *тремя и более детьми* - **0.15%** (31 случай).


А если более детализировать список заемщиков по количеству детей (от 0 до 5), то можно увидеть, что у заемщиков с **пятью детьми** совсем **нет просрочек** по выплатам кредитов:

In [26]:
data_pivot_children_debt = data.pivot_table(index=['children'],\
                                            values = ['debt', 'overdue_loan_percentage'] , aggfunc='sum')

print()
print('Анализ наличия зависимости между количеством детей и возвратом кредита в срок:')
display(data_pivot_children_debt)


Анализ наличия зависимости между количеством детей и возвратом кредита в срок:


,debt,overdue_loan_percentage
children,,
0,1063,4.983358
1,444,2.081478
2,194,0.909474
3,27,0.126576
4,4,0.018752
5,0,0.000000


***Таким образом, можно сделать вывод, что возврат кредита в срок напрямую зависит от количества детей у заемщика***.

***Заемщики с детьми имеют более низкий процент невозврата кредита, чем заемщики без детей.
К тому же, чем больше у заемщика детей, тем меньше случаев просроченной задолженности.***

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Для определения зависимости между семейным положением и возвратом кредита в срок, сформируем сводную таблицу **data_pivot_famstatus_debt** методом *pivot_table()*, используя следующие показатели:

`family_status` — семейное положение;

`debt` — имел ли заемщик задолженность по возврату кредитов (0 - нет, 1 - да);

`overdue_loan_percentage` - процент просроченной задолженности по кредиту


In [27]:
#Cоздаем сводную таблицу `data_pivot_famstatus_debt` с данными о семейном положении заемщика и просрочках:
data_pivot_famstatus_debt = data.pivot_table(index=['family_status'], \
                                             values=['debt', 'overdue_loan_percentage'], aggfunc='sum')

#Применим к data_pivot_famstatus_debt (столбцу debt) сортировку по убыванию (ascending = False):
data_pivot_famstatus_debt = data_pivot_famstatus_debt.sort_values(by='debt', ascending=False)

Для более полного анализа заведем переменную `debt_sum` с **суммой всех случаев просрочки кредитов** 
по категориям столбца `family_status`: 

In [28]:
debt_sum = data_pivot_famstatus_debt['debt'].sum()

Рассчитаем величину просрочек у групп заемщиков в столбце `family_status`, исключая доминирующую категорию (максимальную величину) и долю просрочек в преобладающей категории "женат / замужем". Это необходимо для дальнейшего сравнения и помощи в формировании более развернутых и содержательных выводов.

In [29]:
# рассчитаем сумму всех просрочек по категриям в "family_status", исключая максимальную величину:
debt_difference = debt_sum - data_pivot_famstatus_debt['debt'].max()

# найдем долю просрочек в преобладающей категории "женат / замужем" в %:
share_debt = (data_pivot_famstatus_debt['debt'].max() / debt_sum) * 100

print()
print('Анализ наличия зависимости между семейным положением и возвратом кредита в срок:')

display(data_pivot_famstatus_debt) 
print()
print('Итого просрочек по категориям "family_status":',debt_sum)
print('Сумма просрочек за исключением "женат / замужем":', debt_difference )
print('Доля просрочек категории "женат / замужем":', share_debt )


Анализ наличия зависимости между семейным положением и возвратом кредита в срок:


,debt,overdue_loan_percentage
family_status,,
женат / замужем,927,4.345788
гражданский брак,385,1.804885
Не женат / не замужем,273,1.279827
в разводе,84,0.393793
вдовец / вдова,63,0.295345



Итого просрочек по категориям "family_status": 1732
Сумма просрочек за исключением "женат / замужем": 805
Доля просрочек категории "женат / замужем": 53.52193995381063


Как видно из таблицы *data_pivot_famstatus_debt* **заемщики в статусе "женат / замужем"** на `3.96%`  **чаще просрочивают платежи по кредитам** по сравнению с заемщиками, находящимися в **разводе** (процент просрочки по кредиту - 0.39%) и на `4.05%` по сравнению с заемщиками в статусе **"вдовец / вдова"**(процент просрочки по кредиту - 0.30%)

То есть, общее количество заемщиков с просроченной кредиторской задолженностью, находящихся в официальном браке, превышает суммарное количество всех остальных категорий заемщиков: 

- женат / замужем - **927 случаев** просрочки;

- остальные категории family_status - **805 случаев** просрочки.

**Удельный вес просрочек категории "женат / замужем"** в отношении всех просрочек по категориям "family_status" -  **53.52%**.


*Даже если сгруппировать всех заемщиков категории "family_status" по группам*: "в браке", "холост", "вдова/вдовец" - ситуация не изменится. **Зависимость просроченных платежей от статуса семейного положения сохранится**:


In [30]:
# c помощью функции family_status_group(family_status) распределим всех заемщиков по группам (семейному положению)
def family_status_group(family_status):
    """
    Возвращает статус семьи по значению family_status заемщика, используя правила:
    - 'в браке' при значении family_status -  'женат / замужем' или 'гражданский брак' ;
    - 'холостые' при значении family_status - 'Не женат / не замужем' или 'в разводе';
    - 'вдовец / вдова' - во всех остальнных случаях.
    """
    if family_status == 'женат / замужем' or family_status == 'гражданский брак':
        return 'в браке'
    
    if family_status == 'Не женат / не замужем' or family_status == 'в разводе':
        return 'холостые'
    
    return 'вдовец / вдова'

# проверяем, правильно ли отвечает функция на переданные ей данные
family_status = 'Не женат / не замужем'
print(family_status_group(family_status))


# доавляем новый столбец family_status_group в data
data['family_status_group'] = data['family_status'].apply(family_status_group)

# Проверяем, добавился ли новый столбец family_status_group общий датафрейм
print(data.info())

холостые
<class 'pandas.core.frame.DataFrame'>
Int64Index: 21331 entries, 0 to 21524
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   children                 21331 non-null  int64  
 1   days_employed            21331 non-null  float64
 2   dob_years                21331 non-null  int64  
 3   education                21331 non-null  object 
 4   education_id             21331 non-null  int64  
 5   family_status            21331 non-null  object 
 6   family_status_id         21331 non-null  int64  
 7   gender                   21331 non-null  object 
 8   income_type              21331 non-null  object 
 9   debt                     21331 non-null  int64  
 10  total_income             21331 non-null  int32  
 11  purpose                  21331 non-null  object 
 12  total_income_category    21331 non-null  object 
 13  purpose_category         21331 non-null  object 
 14  overdue_loan_

Видим, что в строке с индексом 16 появился новый столбец `family_status_group`.

Формируем обновленную сводную таблицу `data_pivot_famstatus_debt` по cгруппированным заемщикам (группировка - по статусу семейного положения):

In [31]:
data_pivot_famstatus_debt = data.pivot_table(index=['family_status_group'],\
                                             values=['debt', 'overdue_loan_percentage'], aggfunc='sum')

print() 
print('Анализ наличия зависимости между семейным положением и возвратом кредита в срок:')

# выводим на экран в порядке возрастания:
display(data_pivot_famstatus_debt.sort_values(by='debt', ascending=False))


Анализ наличия зависимости между семейным положением и возвратом кредита в срок:


,debt,overdue_loan_percentage
family_status_group,,
в браке,1312,6.150673
холостые,357,1.673621
вдовец / вдова,63,0.295345


*После группировки ситуация сложилась следующим образом*:

Количество просроченных кредитов у заемщиков **в браке** *намного превышает количество просрочек* **у холостых заемщиков** и заемщиков, находящихся в стусе **вдовец / вдова** (на 955 и 1249 случаев  или на 4.48% и 5.85% соответственно).

***Следовательно, можно сделать вывод, что возврат кредита в срок напрямую зависит от семейного положения заемщика.***

***Заемщики в браке (в т.ч. гражданском) имеют более высокие риски невозврата кредита в срок, чем холостые заемщики.Менее всего риску просрочки кредита подвержены заемщики в статусе вдовец / вдова.***



#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Чтобы определить, есть ли зависимость между уровнем дохода и возвратом кредита в срок, сгруппирем всех заемщиков в категории по уровню дохода, используя столбец `total_income` — ежемесячный доход.

При группировке заемщиков, будем использовать критерии по доходам, приведенные в монографии социологического центра РАН в 2023 году ([Статья](https://ngs.ru/text/economics/2023/07/05/72462857/) от 5 июля 2023 г.):

1. **Наименее обеспеченные заемщики** - ежемесячный доход: **менее 1 прожиточного минимума** / 15 453 руб.;
2. **Низкообеспеченные заемщики** - ежемесячный доход: **1-3 прожиточного минимума** / от 15 453 руб. до 46 359 руб.;
3. **Среднеобеспеченные заемщики** - ежемесячный доход: **3-11 прожиточных минимумов** / от 46 359 руб. до 169 983  руб.;
4. **Высокообеспеченные заемщики** - ежемесячный доход: **более 11 прожиточных минимумов**/ выше 169 983  руб.

*Примечание: с 1 января 2024 г. **величина прожиточного минимума** в целом по РФ на душу населения установлена в размере **15 453 руб**.*

C помощью функции `family_status_group(family_status)` распределим всех заемщиков по группам, в зависимости от величины ежемесячного дохода:

In [32]:
def income_group(total_income):
    """
    Возвращает статус заемщика по значению его total_income, используя правила:
    - 'Наименее обеспеченные' при значении total_income < 15 453 руб. ;
    - 'Низкообеспеченные' при значении total_income от 15 453 руб. до 46 359  руб.;
    - 'Среднеобеспеченные' при значении total_income от 46 359 до 169 983 руб.
    - 'Высокообеспеченные' во всех остальнных случаях (выше 169 983).
    """
    if total_income < 15453:
        return 'Наименее обеспеченные'
    if total_income <= 46359:
        return 'Низкообеспеченные'
    if total_income <= 169983:
        return 'Среднеобеспеченные'
    
    return 'Высокообеспеченные'

# проверяем, правильно ли отвечает функция на переданные ей данные
total_income = 35000
print(income_group(total_income))
total_income = 98000
print(income_group(total_income))

#добавляем новый столбец children_group в data
data['income_group'] = data['total_income'].apply(income_group)

# выводим для проверки первые 10 строк столбца data['income_group']
print()
print(data['income_group'].head())

Низкообеспеченные
Среднеобеспеченные

0    Высокообеспеченные
1    Среднеобеспеченные
2    Среднеобеспеченные
3    Высокообеспеченные
4    Среднеобеспеченные
Name: income_group, dtype: object


Теперь, когда все заемщики распределены на группы в зависимости от величины ежемесячного дохода, создаем сводную таблицу `data_pivot_income_debt`. Для анализа, как и прежде, будем использовать данные из столбцов `'debt'`, `'overdue_loan_percentage'` и созданного `'income_group'`:

In [33]:
data_pivot_income_debt = data.pivot_table(index=['income_group'], \
                                             values=['debt', 'overdue_loan_percentage'], aggfunc='sum')

# отсортируем в data_pivot_income_debt по убыванию столбец 'debt' (ascending = False):
data_pivot_income_debt = data_pivot_income_debt.sort_values(by='debt', ascending=False)

print()
print('Анализ наличия зависимости между уровнем дохода и возвратом кредита в срок:')
display(data_pivot_income_debt)


Анализ наличия зависимости между уровнем дохода и возвратом кредита в срок:


,debt,overdue_loan_percentage
income_group,,
Среднеобеспеченные,1141,5.349023
Высокообеспеченные,574,2.690919
Низкообеспеченные,17,0.079696


Из таблицы `data_pivot_income_debt` видно, что **самый высокий уровень риска просрочки платежей характерен для среднеобеспеченных клиентов**: `1141` случай просрочки на `574` случая у высокообеспеченных и `17` случаев у низкообечпеченных заемщиков.

*Низкообеспеченные клиенты имеют **наименьший потенциал риска несвоевременной уплаты платежей** по кредиту*.

Как ни странно, но *клиенты, попадающие в категорию высокообеспеченных*, так же **имеют достаточно высокую вероятность просрочки платежей**. Процент несвоевременной выплаты у этой группы составляет -  `2,69%`.

***Таким образом, можно сдедать вывод, что нет четкой зависимости между уровнем дохода и возвратом кредита, и если такова существует, то рна очень сложна и неоднозначна.***

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

В задание 18 нами была создана функция, сформировавшая на основании данных из столбца `purpose` новый столбец `purpose_category`, в который вошли следующие категории целей кредита:

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

Для проведения анализа, определящего наличие влияния цели кредита на его своевременный возврат, воспользуемся вышеуказанными категориями целей кредита. Соответсвенно, при формировании сводной таблицы будем пользоваться данными из столбца `purpose_category`.

Итак, сформируем сводную таблицу `data_pivot_purpose_debt`. Для анализа, как и прежде,  используем данные из столбцов `debt`, `overdue_loan_percentage` и рассматриваего выше `purpose_category`:

In [34]:
data_pivot_purpose_debt = data.pivot_table(index=['purpose_category'], \
                                             values=['debt', 'overdue_loan_percentage'], aggfunc='sum')

# отсортируем в data_pivot_purpose_debt по убыванию столбец 'debt' (ascending = False):
data_pivot_purpose_debt = data_pivot_purpose_debt.sort_values(by='debt', ascending=False)

print()
print('Анализ наличия зависимости между целью кредита и возвратом кредита в срок:')
display(data_pivot_purpose_debt)


Анализ наличия зависимости между целью кредита и возвратом кредита в срок:


,debt,overdue_loan_percentage
purpose_category,,
операции с недвижимостью,780,3.656650
операции с автомобилем,400,1.875205
получение образования,369,1.729877
проведение свадьбы,183,0.857906


Из таблицы `data_pivot_purpose_debt` видно, что **количество просроченных выплат по кредитам больше у тех категорий заемщиков, у которых более "дорогостоящая" цель кредита**.

>Так у "*операций на недвижимость*"-  **780** несвоевременных выплат;

>у "*операций с автомобилем*" - **400** несвоевременных выплат;

>на "*получение образования*" просрочено **369** платежей, 

>а на "*проведение свадьбы*" - лишь в **183** случаях было зафиксировано нарушение сроков уплаты.


Иными словами, общий процент просрочек у операций с недвижимостью (`3.66%`) превышает общий процент просроченных платежей на образование и свадьбу на `1.92%` и `2.8%` соответственно.

***Следовательно, можно можно заключить, что цель кредита оказывает прямое влияние на своевременное исполнение обязательств. Более высокие суммы кредита увеличивают финансовуе нагрузку на заемщика, что может привести к увеличению риска просрочки платежей и неспособности выполнить обязательства вовремя***.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

**Пропуски в данных** могут возникнуть по разным **причинам**, *это зависит не только от технических процессов при работе с данными, но и от специфики деятельности компании*.

В нашем проекте мы исследуем статистические данные о платёжеспособности клиентов банка, поэтому первой возможной *причиной возникновения пропусков в данных* нужно отметить **отказ заемщика предоставить информацию**. Предлогом отказа может выступать желание клиента соблюдать конфиденциальность, неготовность раскрывать свои личные данные, смена работы, нестабильный источник заработка.

Еще одной причиной, связанной именно с банковской системой, могут стать **устаревшие данные в системе**. Клиент давний, данные, имеющиеся в базе уже давно устарели и не обновлены, либо в момент сбора информации, доход и стаж были не обязательными для заполнения.

Следующей, часто встречающейся причиной возникновения пропусков в данных, могут являтся **ошибки при вводе данных**. Сотрудник мог пропустить какую-то информацию и не отразить ее в анкете клиента, либо допустить описку (так например в задании 7 мы обнаружили у двух типов клиентов(безработные и пенсионеры) аномально большие значения, здесь сотрудник мог неверно ввести стаж, добавив лишние цифры в графу). 
Таким же случайным образом сотрудник мог и удалить данные, не заметив этого (*а, возможно, он работает на удаленке и, когда отлучился налить себе кофе, его котик прошелся по клавиатуре, удалив какие-то графы, либо добавив в данные лишних цифр*).

Необходимо отметить еще одну немаловажную причину - это **технические проблемы при сборе, обработке и выгрузке данных** из различных источников (различные версии ПО, объемные данные,  передача данных между различными системами и прочее).

**При определении возможных причин появления пропусков в данных, важно делать акцент не только на поиск причин, но и на оптимальные и рациональные варианты их решения.** Это может быть  **правильный выбор метода заполнения пропусков**  медианными или средними  величинами в зависимости от переменных, **использование внешних источников информации** (например, в нашем случае могла бы помочь еще и дополнительная информация о зарплате в определенных профессиях, определенного региона, если бы банку требовался более точный и тщательный анализ).

Так же **не стоить забывать и о коммуникации с клиентами** (можно попытаться собрать дополнительную информацию у самого клиента). Только здесь **важно не прибегать к "навящимому сервису"**, так как это может, как минимум, настроить клиента против сотрудника банка, как максимум, дать соврешенно обратный эффект - клиент может задуматься об отказе услуг "давящего" банка).


***Вывод: оперативное и, по возможности, максимально точное выявление вероятных причин появления пропусков в исходных данных, а так же их рациональное и "тонкое" решение, помогут не только улучшить качество анализа и принятия решений в исследуемом банке, но и будут способствовать повышению эффективности его операций и улучшению взаимоотношений с клиентами.***

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

Перед тем, как объяснить, почему пропуски для количественных переменных лучше заполнять медианным значением, необходимо напомнить, что в целом **пропуски заполняются либо медианой, либо средним арифметическим значением**.

**Отличие медианного значения от среднего арифметического в способе нахождения характерного значения**.

Если **среднее арифметическое** — это сумма всех значений, поделённая на количество значений, то **медиана** — это такое число в выборке, что ровно половина элементов больше него, а другая половина — меньше.

Заполнение пропусков в колличественных значениях медианой наиболее предпочтительно, так как среднее значение некорректно характеризует данные, в особенности, когда некоторые показатели сильно выделяются среди большинства. Происходит завышение/занижение показателей (в зависимости от ситуации), и как следствие, искажение данных.



В задании 7 мы для каждого типа занятости выводили **медианное значение** трудового стажа `days_employed`:

In [35]:
print(data.groupby('income_type')['days_employed'].median())

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.137274
компаньон            1555.947387
пенсионер          365286.622650
предприниматель       520.848083
сотрудник            1572.328285
студент               578.751554
Name: days_employed, dtype: float64


Напомним,что у двух типов (безработные и пенсионеры) получились аномально большие значения. Исправить такие значения сложно, поэтому мы оставили их как есть.

Теперь попробуем для каждого типа занятости вывести **среднее арифметическое значение** трудового стажа `days_employed`:

In [36]:
print(data.groupby('income_type')['days_employed'].mean())

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          3332.636454
компаньон            2062.884254
пенсионер          365051.368541
предприниматель       520.848083
сотрудник            2250.942428
студент               578.751554
Name: days_employed, dtype: float64


У следующих категорий: `госслужащий`, `компаньон`, `сотрудник` среднее арифметического значение трудового стажа больше, чем медианное значение.  Это подтверждает вышесказанное и говорит о том, что среднее арифметическое в количественных значениях существенно искажает распределение.

Теперь попробуем этим же способом для каждого типа занятости вывести и сравнить медиану и среднее значение уровня ежемесячного дохода.

**Медианное значение** дохода `total_income`:

In [37]:
print(data.groupby('income_type')['total_income'].median())

income_type
безработный        131339
в декрете           53829
госслужащий        150447
компаньон          172357
пенсионер          118514
предприниматель    499163
сотрудник          142594
студент             98201
Name: total_income, dtype: int32


**Среднее арифметическое** значение дохода `total_income`:

In [38]:
print(data.groupby('income_type')['total_income'].mean())

income_type
безработный        131339.000000
в декрете           53829.000000
госслужащий        168728.454859
компаньон          199608.615811
пенсионер          135320.753673
предприниматель    499163.000000
сотрудник          159547.951975
студент             98201.000000
Name: total_income, dtype: float64


К категориям, у которых изменились показатели при расчете среднего значения: `госслужащий`, `компаньон`, `сотрудник`, добавилась еще одна - `пенсионер`. 
Как и при расчете показателей стажа, среднее значение уровня дохода выше, чем медиана. 
Это вновь говорит о том, что медиана, в отличие от среднего значения, отдает более точные значения. А это, в свою очередь, дает возможность аналитику работать с наиболее реальными данными, что улучшает качество подготовленного им, по результатам анализа, заключения.

***Таким образом, можно сделать вывод, что заполнение пропусков медианным значением — лучшее решение для количественных переменных. Изменения в данных имеют меньший эффект на медиану, что придает ей стабильность и защищает от воздействия случайных факторов или изменений.***

### Шаг 4: общий вывод.

В ходе настоящего проекта **был проведен анализ статистических данных платежеспособности клиентов банка с целью** выяления ключевых факторов, влияющих на способность потенциальных заемщиков выплачивать кредиты вовремя. А именно, необходимо было **разобраться влияет ли семейное положение, уровень дохода и количество детей клиента на факт погашения кредита в срок**. 

**Анализ был необходим** для обеспечения более глубокого понимания кредитного поведения заемщиков с целью предоставления банку результатов исследования, которые будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

**Ниже предоставлен краткий обзор проведённой работы**.

**В процессе предобработкив в данных были обнаружены** пропущенные значения в столбцах `days_employed` и `total_income`, явные, неявные дубликаты и артефакты (аномалии).

**Чтобы решить эти проблемы** пришлось:
- заполнить пропуски медианными значениями для каждой категории заемщиков; 
- обработать неявные дубликаты, удалить явные;
- заменить вещественный тип данных на целочисленный;
- обработать все артефакты (аномалии) данных, а именно, заменить все отрицательные значения положительными и удалить два аномальных значения.

**Далее все клиенты были поделены на категории в зависимости от уровня дохода и целей кредита**. По этим категориям были созданы два отдельных столбца и добавлены в исходный датафрейм.

**При оперделении наличия зависимости между количеством детей и возвратом кредита в срок** было добавлено еще два столбца: общий процент просроченной задолженности по кредиту по каждому заемщику (при группировке заемщиков по категориям, можно увидеть какой процент просрочек из общего количество занимает именно эта категория) и  статус семьи заемщика, в зависимости от количества детей.

В ходе анализа была установлена следующая зависимость:  ***возврат кредита в срок обратно пропорционально зависит от количества детей у заемщика. Заемщики с детьми имеют более низкий процент невозврата кредита, чем заемщики без детей и чем больше у заемщика детей, тем меньше случаев просроченной задолженности***.

**Далее нужно было определить существование зависимости между семейным положением и возвратом кредита в срок**. Помимо формирования сводной таблицы из данных о семейном положении заемщика и количеством просрочек, для наиболее полного отражения кредитной картины были найдены еще такие показатели, как итоговая сумма всех просрочек по категориям "family_status", сумма просрочек за исключением "женат / замужем" и доля просрочек преобладающей категории.

На основании полученных данных был сделан вывод, что ***возврат кредита в срок напрямую зависит от семейного положения заемщика.Заемщики в браке (в т.ч. гражданском) имеют более высокие риски невозврата кредита в срок, чем холостые заемщики. Менее всего риску просрочки кредита подвержены заемщики в статусе вдовец / вдова***.

**В процессе определения наличия зависимости между уровнем дохода и возвратом кредита в срок**, было принято решение не пользоваться уже имеющейся сводной таблицей из задания 16, а  сгруппировать всех заемщиков в категории по уровню дохода, пользуясь критериями по доходам, приведенные в монографии социологического центра РАН в 2023 году. Полученные данные позволили сделать вывод, что ***нет четкой зависимости между уровнем дохода и возвратом кредита, и если такова существует, то она очень сложна и неоднозначна***.

**При выявлении возможной зависимости целей кредита на его возврат в срок** для формирования сводной таблицы использовали уже имеющийся столбец из задания 17 с разбивкой заемщиков по целям кредита.

Из анализа данных стало видно, что ***цель кредита напрямую влияет на его своевременный возврат. Более высокие суммы кредита увеличивают финансовуе нагрузку на заемщика, что может привести к увеличению риска просрочки платежей и неспособности выполнить обязательства вовремя***.

Далее были приведены **возможные причины появления пропусков в исходных данных**. 

К ним могут относиться:
- отказ заемщика предоставить информацию;
- устаревшие данные в системе;
- ошибки при вводе данных;
- технические проблемы при сборе, обработке и выгрузке данных.

При определении возможных причин появления пропусков в данных, **была отмечена важность акцентирования внимания не только на поиске причин возникновения пропусков, но и на оптимальных и рациональных вариантах их решения.** Было предложено научиться делать правильный выбор метода заполнения пропусков медианными или средними  величинами в зависимости от переменных, использование внешних источников информации, коммуникации с клиентами, не прибегая к "навящимому сервису".

Затем, при объяснении важности заполнения пропусков для количественных переменных именно медианным значением, были даны определения медиане и среднему арифметическому значению, проведен сравнительный анализ выводимых медианных и средних значений трудового стажа и уровня дохода. 

На примерах стало наглядно видно, что:

- средняя величина завышает показатели по некоторым категориям сотрудников;
- медиана, в отличие от среднего значения, отдает более точную величину;
- среднее значение некорректно характеризует данные, в особенности, когда некоторые показатели сильно выделяются среди большинства;
- изменения в данных имеют меньший эффект на медиану, что придает ей стабильность и защищает от воздействия случайных факторов или изменений.

Соответственно, **заполнение пропусков медианным значением — лучшее решение для количественных переменных**.

***В итоге, хочется надеяться, что полученные результаты исследования окажутся ценными при формировании модели кредитного скоринга — инструмента, на основе которого оценивается способность потенциального заёмщика вернуть кредит банку. Надеемся, что эти выводы станут полезным вкладом в дальнейшее совершенствование процессов кредитования, снижение рисков и обеспечение более эффективной работы банка.***
